editor/src/components/editor/actions/actions.tsx

```diff
@@ -5241,80 +5241,128 @@ export const UPDATE_FNS = {
-              (parentElement) => {
-                if (isJSXElement(parentElement)) {
-                  insertedElementChildren.push(...parentElement.children)
-                  return jsxElement(parentElement.name, parentElement.uid, parentElement.props, [])
-                } else {
-                  throw new Error(`Not handled yet.`)
-                }
-              },
+              element,
+              withMaybeUpdatedParent,
+              action.indexPosition,
+            )
+            detailsOfUpdate = withInsertedElement.insertionDetails
+
+            const newPath = EP.appendToPath(action.targetParent, newUID)
+            newSelectedViews.push(newPath)
+          } else if (action.toInsert.element.type === 'JSX_CONDITIONAL_EXPRESSION') {
+            const element = jsxConditionalExpression(
+              newUID,
+              action.toInsert.element.condition,
+              action.toInsert.element.originalConditionString,
+              action.toInsert.element.whenTrue,
+              action.toInsert.element.whenFalse,
+              action.toInsert.element.comments,
             )
-          }
 
-          insertedElementChildren.push(...action.toInsert.element.children)
-          const element = jsxElement(insertedElementName, newUID, props, insertedElementChildren)
+            withInsertedElement = insertElementAtPath(
+              editor.projectContents,
+              openFilename,
+              action.targetParent,
+              element,
+              utopiaComponents,
+              action.indexPosition,
+            )
+            detailsOfUpdate = withInsertedElement.insertionDetails
+
+            const newPath = EP.appendToPath(action.targetParent, newUID)
+            newSelectedViews.push(newPath)
+          } else if (action.toInsert.element.type === 'JSX_FRAGMENT') {
+            const element = jsxFragment(
+              newUID,
+              action.toInsert.element.children,
+              action.toInsert.element.longForm,
+            )
 
-          const withInsertedElement = insertElementAtPath(
-            editor.projectContents,
-            openFilename,
-            action.targetParent,
-            element,
-            withMaybeUpdatedParent,
-            action.indexPosition,
-          )
-          detailsOfUpdate = withInsertedElement.insertionDetails
+            withInsertedElement = insertElementAtPath(
+              editor.projectContents,
+              openFilename,
+              action.targetParent,
+              element,
+              utopiaComponents,
+              action.indexPosition,
+            )
+            detailsOfUpdate = withInsertedElement.insertionDetails
 
-          const newPath = EP.appendToPath(action.targetParent, newUID)
-          newSelectedViews.push(newPath)
+            const newPath = EP.appendToPath(action.targetParent, newUID)
+            newSelectedViews.push(newPath)
+          } else {
+            assertNever(action.toInsert.element)
```

Comment:
Will TS catch new types of `action.toInsert.element.type` with this? Or is this a runtime check?

Reply:
this will cause a compile-time error as long as the last `else` is there, and a case of `action.toInsert.element.type` isn't handled (an `if/else if/else` chain like this is essentially the same as a switch/case/default with no fall-throughs).